recipes/nanomsg/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conan import ConanFile
+from conan.tools.files import apply_conandata_patches,  get, copy, rm, rmdir
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+class NanomsgConan(ConanFile):
+    name = "nanomsg"
+    description = "a socket library that provides several common communication patterns."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nanomsg/nanomsg"
+    topics = ("socket", "protocols", "communication")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_coverage": [True, False],
+        "enable_getaddrinfo_a":[True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enable_coverage": False,
+        "enable_getaddrinfo_a":True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["NN_STATIC_LIB"] = not self.options.shared
+        tc.variables["NN_ENABLE_COVERAGE"] = self.options.enable_coverage
+        tc.variables["NN_ENABLE_GETADDRINFO_A"] = self.options.enable_getaddrinfo_a
```

Comment:
```suggestion
        tc.variables["NN_ENABLE_GETADDRINFO_A"] = self.options.enable_getaddrinfo_a
        tc.variables["NN_ENABLE_DOC"] = False
        tc.variables["NN_TESTS"] = False
        tc.variables["NN_TOOLS"] = self.options.enable_tools
```
I would recommend disabling options that will not be used for packaging and consumer even more time for building.

https://github.com/nanomsg/nanomsg/blob/1.2/CMakeLists.txt#LL100C22

Plus, I would add an option to build or not tools, as most of cases, users want to consume the library only.

Reply:
I will add build_tools option and set default value False.